src/mlpack/methods/ann/gan/gan_impl.hpp

```diff
@@ -137,15 +120,54 @@ GAN<Model, InitializationRuleType, Noise, PolicyType>::GAN(
     clippingParameter(network.clippingParameter),
     lambda(network.lambda),
     reset(network.reset),
-    counter(network.counter),
     currentBatch(network.currentBatch),
     parameter(std::move(network.parameter)),
     numFunctions(network.numFunctions),
-    noise(std::move(network.noise))
+    noise(std::move(network.noise)),
+    deterministic(network.deterministic),
+    genWeights(network.genWeights),
+    discWeights(network.discWeights)
 {
   /* Nothing to do here */
 }
 
+template<
+  typename Model,
+  typename InitializationRuleType,
+  typename Noise,
+  typename PolicyType
+>
+void GAN<Model, InitializationRuleType, Noise, PolicyType>::ResetData(
+    arma::mat trainData)
+{
+  currentBatch = 0;
+
+  numFunctions = trainData.n_cols;
+  noise.set_size(noiseDim, batchSize);
+
+  deterministic = true;
+  ResetDeterministic();
+
+  this->predictors.set_size(trainData.n_rows, numFunctions + batchSize);
+  this->predictors.cols(0, numFunctions - 1) = std::move(trainData);
```

Comment:
Are you sure that this is right?  The number of columns in `predictors` is `numFunctions + batchSize`, but I see in this line here we set the first `numFunctions` columns; what about the additional `batchSize` columns?  If they're not necessary, maybe it's better to just do `this->predictors = std::move(trainData);` (which will also be more efficient as no copy will be needed).

Reply:
This is quite correct. The additional `batchsize` input is taken from generator while training.
For e.g you can check line `264, 268, 339, 341,` to know how it is used.

Reply:
Ok, I see.  It might be worth adding a comment above the `set_size()` to reassure paranoid readers like me that the memory will eventually be set correctly. :)

Reply:
Yes, Done. :+1: